Source/Core/InputCommon/ControllerEmu/ControlGroup/Cursor.cpp

```diff
@@ -82,15 +82,30 @@ ControlState Cursor::GetGateRadiusAtAngle(double ang) const
 
 Cursor::StateData Cursor::GetState(const bool adjusted)
 {
-  if (!adjusted)
-  {
-    const auto raw_input = GetReshapableState(false);
+  const ReshapeData input = GetReshapableState(adjusted);
+  const StateData state = adjusted ? UpdateState(input) : StateData{input.x, input.y};
+  return state;
+}
 
-    return {raw_input.x, raw_input.y};
-  }
+Cursor::StateData Cursor::GetState(const bool adjusted,
+                                   const ControllerEmu::InputOverrideFunction& override_func)
+{
+  if (!override_func)
+    return GetState(adjusted);
 
-  const auto input = GetReshapableState(true);
+  const ReshapeData input = GetReshapableState(adjusted);
+  StateData state = adjusted ? UpdateState(input) : StateData{input.x, input.y};
```

Comment:
Can't this be simplified to

```
StateData state = GetState(adjusted);
if (!override_func)
  return state;
```

? Same for other inputs.

Reply:
I've applied this simplification to AnalogStick and Cursor, but for the other ones it wouldn't apply cleanly due to GetState doing extra processing after the point where the override function would be called.